ext/cl_khr_subgroup_extensions.asciidoc

```diff
@@ -0,0 +1,1072 @@
+For the functions below, the generic type name `gentype` may be one of the supported built-in scalar data types `char`, `uchar`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `float`, `double` (if double precision is supported), or `half` (if half precision is supported).
+
+[cols="3a,2",options="header",]
+|=======================================================================
+|*Function*
+|*Description*
+
+|[source,c]
+----
+gentype sub_group_non_uniform_reduce_add(
+    gentype value )
+gentype sub_group_non_uniform_reduce_min(
+    gentype value )
+gentype sub_group_non_uniform_reduce_max(
+    gentype value )
+gentype sub_group_non_uniform_reduce_mul(
+    gentype value )
+----
+| Returns the summation, multiplication, minimum, or maximum of _value_ for all active work items in the subgroup.
+
+Note: This behavior is the same as the *add*, *min*, and *max* reduction built-in functions from `cl_khr_subgroups` and OpenCL 2.1, except these functions support additional types and need not be encountered by all work items in the subgroup executing the kernel.
+
+|[source,c]
+----
+gentype sub_group_non_uniform_scan_inclusive_add(
+    gentype value )
+gentype sub_group_non_uniform_scan_inclusive_min(
+    gentype value )
+gentype sub_group_non_uniform_scan_inclusive_max(
+    gentype value )
+gentype sub_group_non_uniform_scan_inclusive_mul(
+    gentype value )
+----
+| Returns the result of an inclusive scan operation, which is the summation, multiplication, minimum, or maximum of _value_ for all active work items in the subgroup with a subgroup local ID less than or equal to this work item's subgroup local ID.
+
+Note: This behavior is the same as the *add*, *min*, and *max* inclusive scan built-in functions from `cl_khr_subgroups` and OpenCL 2.1, except these functions support additional types and need not be encountered by all work items in the subgroup executing the kernel.
+
+|[source,c]
+----
+gentype sub_group_non_uniform_scan_exclusive_add(
+    gentype value )
+gentype sub_group_non_uniform_scan_exclusive_min(
+    gentype value )
+gentype sub_group_non_uniform_scan_exclusive_max(
+    gentype value )
+gentype sub_group_non_uniform_scan_exclusive_mul(
+    gentype value )
+----
+| Returns the result of an exclusive scan operation, which is the summation, multiplication, minimum, or maximum of _value_ for all active work items in the subgroup with a subgroup local ID less than this work item's subgroup local ID.
+
+If there is no active work item in the subgroup with a subgroup local ID less than this work item's subgroup local ID then an identity value `I` is returned.
+For *add*, the identity value is `0`.
+For *min*, the identity value is the largest representable value for integer types, or `+INF` for floating point types.
+For *max*, the identity value is the smallest representable value for integer types, or `-INF` for floating point types.
```

Comment:
Not sure if it's just me but I read "smallest representable value" as magnitude, which would be `0` for signed or unsigned types.  Maybe "minimum" is less ambiguous?

```suggestion
For *max*, the identity value is the minimum representable value for integer types, or `-INF` for floating point types.
```

Reply:
Can always come back and change this later though.